cot/src/form/types.rs

```diff
@@ -0,0 +1,383 @@
+        &self.0
+    }
+
+    /// Consumes the object and returns the password as a string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::form::types::Password;
+    ///
+    /// let password = Password::new("password");
+    /// assert_eq!(password.into_string(), "password");
+    /// ```
+    #[must_use]
+    pub fn into_string(self) -> String {
+        self.0
+    }
+}
+
+impl From<&Password> for Password {
+    fn from(password: &Password) -> Self {
+        password.clone()
+    }
+}
+
+impl From<&str> for Password {
+    fn from(password: &str) -> Self {
+        Self::new(password)
+    }
+}
+
+impl From<String> for Password {
+    fn from(password: String) -> Self {
+        Self::new(password)
+    }
+}
+
+/// A validated email address.
+///
+/// This is a newtype wrapper around
+/// [`EmailAddress`](email_address::EmailAddress) that provides validation and
+/// integration with Cot's database system. It ensures email addresses
+/// comply with RFC 5321/5322 standards.
+///
+/// # Examples
+///
+/// ```
+/// use std::str::FromStr;
+///
+/// use cot::form::types::Email;
+///
+/// // Parse from a string
+/// let email = Email::from_str("user@example.com").unwrap();
+///
+/// // Convert using TryFrom
+/// let email = Email::try_from("user@example.com").unwrap();
+/// ```
+#[derive(Clone, Debug)]
+pub struct Email(EmailAddress);
+
+impl Email {
+    /// Creates a new `Email` from a string, validating that it's a proper email
+    /// address.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the email address is invalid according to RFC
+    /// standards.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::form::types::Email;
+    ///
+    /// let email = Email::new("user@example.com").unwrap();
+    /// assert!(Email::new("invalid").is_err());
+    /// ```
+    pub fn new<S: AsRef<str>>(email: S) -> Result<Email, email_address::Error> {
+        EmailAddress::from_str(email.as_ref()).map(Self)
+    }
+
+    /// Returns the email address as a string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::str::FromStr;
+    ///
+    /// use cot::form::types::Email;
+    ///
+    /// let email = Email::from_str("user@example.com").unwrap();
+    /// assert_eq!(email.as_str(), "user@example.com");
+    /// ```
+    #[must_use]
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+
+    /// Returns a reference to the underlying `EmailAddress`.
+    ///
+    /// This is useful when you need to access functionality provided by the
+    /// `email_address` crate directly.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::str::FromStr;
+    ///
+    /// use cot::form::types::Email;
+    ///
+    /// let email = Email::from_str("user@example.com").unwrap();
+    /// let domain = email.as_inner().domain();
+    /// assert_eq!(domain, "example.com");
+    /// ```
+    #[must_use]
+    pub fn as_inner(&self) -> &EmailAddress {
```

Comment:
Do you think it makes sense to remove this `as_inner` method and instead expose the methods that could be potentially useful for users (like `domain()`, `local_part()`, etc.) instead? We generally try not to expose crates we depend on unnecessarily, as it makes it harder to swap these crates for different ones in the future, or even upgrade their versions (because every major version bump of an exposed crate results in a breaking change for us, too).

Reply:
@seqre I ended up going with the first approach, I'm still unsure what the idiomatic approach is, or if the second approach is a genuine concern. If it isn't, let me know and I'll be happy to go in that direction.